sys/net/sock/tls/tcp_tls.c

```diff
@@ -0,0 +1,263 @@
+            continue;
+        }
+
+        if (ret == -ETIMEDOUT) {
+            printf("TCP Read timeout after %d ms\n", (int)TLS_DEFAULT_TIMEOUT);
+            tries++;
+            if (total_received > 0) return total_received;
+            if (tries >= MAX_RETRIES) return WOLFSSL_CBIO_ERR_TIMEOUT;
+            continue;
+        }
+
+        if (ret < 0) {
+            printf("TCP Read error: %d\n", ret);
+            if (total_received > 0) return total_received;
+            return WOLFSSL_CBIO_ERR_GENERAL;
+        }
+    }
+    return total_received > 0 ? total_received : WOLFSSL_CBIO_ERR_GENERAL;
+}
+
+static int _wolfssl_tcp_send(WOLFSSL* ssl, char* buf, int sz, void* ctx) {
+    sock_tls_tcp_t *sock = (sock_tls_tcp_t *)ctx;
+    int total_sent = 0;
+    int remaining = sz;
+    int tries = 0;
+
+    while (remaining > 0 && tries < MAX_RETRIES) {
+        int chunk_size = (remaining > CHUNK_SIZE) ? CHUNK_SIZE : remaining;
+        int ret = sock_tcp_write(sock->tcp_sock, buf + total_sent, chunk_size);
+
+        if (ret > 0) {
+            total_sent += ret;
+            remaining -= ret;
+            tries = 0;
+            continue;
+        }
+
+        if (ret == -EAGAIN || ret == -EWOULDBLOCK) {
+            tries++;
+            if (tries >= MAX_RETRIES) {
+                printf("TCP Send failed after %d tries\n", MAX_RETRIES);
+                if (total_sent > 0) return total_sent;
+                return WOLFSSL_CBIO_ERR_GENERAL;
+            }
+            ztimer_sleep(ZTIMER_MSEC, 100 * tries);
+            continue;
+        }
+
+        if (ret < 0) {
+            if (total_sent > 0) return total_sent;
+            return WOLFSSL_CBIO_ERR_GENERAL;
+        }
+    }
+    return total_sent > 0 ? total_sent : WOLFSSL_CBIO_ERR_GENERAL;
+}
+
+int sock_tls_tcp_create(sock_tls_tcp_t *sock, WOLFSSL_METHOD *method, const unsigned char *ca_cert, unsigned int ca_cert_len) {
+    if (!sock || !method) return -EINVAL;
+
+    memset(sock, 0, sizeof(sock_tls_tcp_t));
+    sock->ctx = wolfSSL_CTX_new(method);
+    if (!sock->ctx) return -ENOMEM;
+
+    if (wolfSSL_CTX_set_cipher_list(sock->ctx, "AES128-SHA256") != SSL_SUCCESS) {
+        wolfSSL_CTX_free(sock->ctx);
+        return -EINVAL;
+    }
+
+    if (wolfSSL_CTX_load_verify_buffer(sock->ctx, ca_cert, ca_cert_len, SSL_FILETYPE_PEM) != SSL_SUCCESS) {
+        wolfSSL_CTX_free(sock->ctx);
+        return -EINVAL;
+    }
+
+    wolfSSL_SetIORecv(sock->ctx, _wolfssl_tcp_receive);
+    wolfSSL_SetIOSend(sock->ctx, _wolfssl_tcp_send);
+
+    sock->ssl = wolfSSL_new(sock->ctx);
+    if (!sock->ssl) {
+        wolfSSL_CTX_free(sock->ctx);
+        sock->ctx = NULL;
+        return -ENOMEM;
+    }
+
+    wolfSSL_set_verify(sock->ssl, SSL_VERIFY_PEER, NULL);
+    return 0;
+}
+
+int sock_tls_tcp_connect(sock_tls_tcp_t *sock, const sock_tcp_ep_t *remote, uint16_t local_port, uint16_t flags) {
+    if (!sock || !remote) return -EINVAL;
+
+    sock->tcp_sock = malloc(sizeof(sock_tcp_t));
```

Comment:
Can we get rid of the dynamic memory allocation? Maybe introduce some internal state for that.